resources/patches/ruby21/0001-Enable-side-by-side-assembly.patch

```diff
@@ -0,0 +1,84 @@
+diff --git a/win32/resource.rb b/win32/resource.rb
+index 786edb0..9236cad 100755
+--- a/win32/resource.rb
++++ b/win32/resource.rb
+@@ -52,6 +52,7 @@ end
+   [$rubyw_name,  CONFIG["EXEEXT"], 'VFT_APP', 'GUI', rubyw_icon || ruby_icon],
+   [$so_name,     '.dll',           'VFT_DLL', 'DLL', dll_icons.join],
+ ].each do |base, ext, type, desc, icon|
++  manifest = (type=='VFT_APP') ? "1" : "2"
+   open(base + '.rc', "w") { |f|
+     f.binmode if /mingw/ =~ RUBY_PLATFORM
+ 
+@@ -59,8 +60,10 @@ end
+ #ifndef __BORLANDC__
+ #include <windows.h>
+ #include <winver.h>
++#include <winuser.h>
+ #endif
+ 
++#{manifest} RT_MANIFEST "ruby.bin.manifest"
+ #{icon || ''}
+ VS_VERSION_INFO VERSIONINFO
+  FILEVERSION    #{nversion}
+@@ -95,3 +98,60 @@ EOF
+   }
+ end
+ 
++manifest_version = RUBY_VERSION + ".0"
++manifest_arch = (CONFIG["target_cpu"] == "x64") ? 'amd64' :
++  (CONFIG["target_cpu"] =~ /i[3456]86/) ? 'x86' :
++  'error'
++
++open('ruby.bin.manifest', "w") { |f|
++    f.binmode if /mingw/ =~ RUBY_PLATFORM
++
++    f.print <<EOF_MANI_BIN
++<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
++
++  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
++      <application>
++        <!-- Windows Vista -->
++          <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}"/>
++        <!-- Windows 7 -->
++          <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}"/>
++        <!-- Windows 8 -->
++          <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}"/>
++        <!-- Windows 8.1 -->
++          <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}"/>
++        <!-- Windows 10 -->
++          <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}"/>
++      </application>
++  </compatibility>
```

Comment:
Thank you for the code review.
I've found that the OS compatibility element is unnecessary for a private assembly.

See https://msdn.microsoft.com/en-us/library/windows/desktop/aa375150(v=vs.85).aspx

I tested building a ruby and confirmed that all test pass.

I chose patching the win32/resource.rb
because its run creates each resource file(*.rc) for ruby.exe, rubyw.exe and x64-msvcrt-ruby*.dll
during "make all" invoked sequence.

I think the file win32/resource.rb is stable enough and patching the file is easier and better outlook than modifying official makefile here and there.

If there is something wrong with the file, patching fails properly.

Though I don't against changing the way to create the private assembly resource files.
I just don't come up with another good solution.